dev_docs/kibana_platform_plugin_intro.mdx

```diff
@@ -0,0 +1,271 @@
+
+The table below explains how each lifecycle relates to the state of Kibana.
+
+| lifecycle	| purpose |	server |	browser |
+| ---------- | ------ | ------- | ----- |
+| setup | perform "registration" work to setup environment for runtime |configure REST API endpoint, register saved object types, etc. | configure application routes in SPA, register custom UI elements in extension points, etc. |
+| start | bootstrap runtime logic | respond to an incoming request, request Elasticsearch server, etc. | start polling Kibana server, update DOM tree in response to user interactions, etc.| 
+| stop | cleanup runtime | dispose of active handles before the server shutdown. | store session data in the LocalStorage when the user navigates away from Kibana, etc. |
+
+Different service interfaces can and will be passed to setup, start, and stop because certain functionality makes sense in the context of a running plugin while other types 
+of functionality may have restrictions or may only make sense in the context of a plugin that is stopping.
+
+## How plugin's interact with each other, and Core
+
+The lifecycle-specific contracts exposed by core services are always passed as the first argument to the equivalent lifecycle function in a plugin. 
+For example, the core http service exposes a function createRouter to all plugin setup functions. To use this function to register an HTTP route handler, 
+a plugin just accesses it off of the first argument:
+
+```ts
+import type { CoreSetup } from 'kibana/server';
+
+export class MyPlugin {
+  public setup(core: CoreSetup) {
+    const router = core.http.createRouter();
+    // handler is called when '/path' resource is requested with `GET` method
+    router.get({ path: '/path', validate: false }, (context, req, res) => res.ok({ content: 'ok' }));
+  }
+}
+```
+
+Unlike core, capabilities exposed by plugins are not automatically injected into all plugins.
+Instead, if a plugin wishes to use the public interface provided by another plugin, it must first declare that plugin as a
+ dependency in it’s kibana.json manifest file.
+
+** demo kibana.json**
+
+```
+{
+  "id": "demo",
+  "requiredPlugins": ["foobar"],
+  "server": true,
+  "ui": true
+}
+```
+
+With that specified in the plugin manifest, the appropriate interfaces are then available via the second argument of setup and/or start:
+
+```ts
+import type { CoreSetup, CoreStart } from 'kibana/server';
+import type { FoobarPluginSetup, FoobarPluginStart } from '../../foobar/server';
+
+interface DemoSetupPlugins {  [1]
+  foobar: FoobarPluginSetup;
+}
+
+interface DemoStartPlugins {
+  foobar: FoobarPluginStart;
+}
+
+export class AnotherPlugin {
+  public setup(core: CoreSetup, plugins: DemoSetupPlugins) { [2]
+    const { foobar } = plugins;
+    foobar.getFoo(); // 'foo'
+    foobar.getBar(); // throws because getBar does not exist
```

Comment:
What is the point of showing this?  It's unclear from the previous description why `getBar()` doesn't exist.

Reply:
great catch!! Bad copy/paste from https://www.elastic.co/guide/en/kibana/current/kibana-platform-plugin-api.html, I missed some content.